dlpy/model.py

```diff
@@ -268,14 +268,19 @@ def fit(self, data, inputs=None, target=None, data_specs=None, mini_batch_size=1
         if target is None and '_label_' in input_table.columns.tolist():
             target = '_label_'
 
-        if self.model_weights.to_table_params()['name'].upper() in \
+        # check whether the field is none or not
+        if self.model_weights is not None and self.model_weights.to_table_params()['name'].upper() in \
                 list(self._retrieve_('table.tableinfo').TableInfo.Name):
             print('NOTE: Training based on existing weights.')
             init_weights = self.model_weights
         else:
             print('NOTE: Training from scratch.')
             init_weights = None
 
+        # when model_weights is none, reset it
+        if self.model_weights is None:
+            self.model_weights = self.conn.CASTable('{}_weights'.format(self.model_name))
+
         if save_best_weights and self.best_weights is None:
             self.best_weights = random_name('model_best_weights', 6)
 
```

Comment:
it is quite difficult to see your changes, especially when you change the location of the functions :) github thinks you removed everything and added a bunch of things later.